HealthMonitoring.SelfHost/Controllers/EndpointsController.cs

```diff
@@ -19,6 +20,7 @@ public class EndpointsController : ApiController
     {
         private readonly IEndpointRegistry _endpointRegistry;
         private readonly IEndpointStatsRepository _endpointStatsRepository;
+        private static readonly ILog Logger = LogManager.GetLogger<EndpointsController>();

```

Comment:
I agree with Will here. We should be using generic request/response logging, added in similar way how exceptions has been added.

It could possibly log the operation that has been called like:
`Request Verb=GET URI=/api/endpoints?filterXXX=YYY`
`Response StatusCode=404 Verb=GET URI=/api/endpoints?filterXXX=YYY`